realm/realm-jni/src/io_realm_internal_TableQuery.cpp

```diff
@@ -1461,7 +1461,7 @@ JNIEXPORT void JNICALL Java_io_realm_internal_TableQuery_nativeIsNull(
         } else {
             switch (col_type) {
                 case type_Link:
-                    pQuery->and_query(src_table_ref->column<Link>(S(column_idx)).is_null());
+                    ThrowException(env, IllegalArgument, "isNull() by nested query for link field is not supported.");
```

Comment:
Instead of doing it like this. Wouldn't it be better to catch the exception from core in CATCH_STD (if it is an explicit exception type) and transform it there. That way it will also be caught when when core actually adds support for it.
See util.cpp (ConvertException)